src/app/shell/admin-tools/data/history-log/history-log-filters/history-log-filters.component.ts

```diff
@@ -1,41 +1,93 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
 import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
 
 import { DropdownData, FilterData } from 'shared/models/history-log.model';
+import { FilterOptions, HistoryLogTypes } from 'shared/enum/history.log';
+import { ProviderAdminOperationOptions } from 'shared/constants/drop-down';
 
 @Component({
   selector: 'app-history-log-filters',
   templateUrl: './history-log-filters.component.html',
   styleUrls: ['./history-log-filters.component.scss']
 })
-export class HistoryLogFiltersComponent implements OnInit {
+export class HistoryLogFiltersComponent implements OnInit, OnChanges {
   @Input() public dropdownOptions: DropdownData;
+  @Input() public tabName: HistoryLogTypes;
   @Output() public filterData = new EventEmitter<FilterData>();
 
   public filtersForm: FormGroup;
+  public formControlName: string = '';
+  public additionalFormControlName: string = '';
+  public additionalDropdownOptions = ProviderAdminOperationOptions;
+
+  private baseCountOfFiltersFormFields = 2;
 
   constructor(private fb: FormBuilder) {}
 
   public ngOnInit(): void {
-    const currentDate = new Date();
-    const monthAgoDate = new Date(currentDate);
-    monthAgoDate.setMonth(currentDate.getMonth() - 1);
-
-    this.filtersForm = this.fb.group({
-      dateFrom: new FormControl(monthAgoDate),
-      dateTo: new FormControl(currentDate),
-      options: new FormControl('')
-    });
-
     this.filterData.emit(this.filtersForm.value);
   }
+  
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.setBaseFiltersForm();
```

Comment:
Do I need to do this every time a change occurs, or only when switching tabs?

Reply:
We need this only when switching tabs, but I can't find better solution. Could you help me with this?

Reply:
We need this only when tabs switching, so I can move this code to the component on higher level. There are mat-tabs in this component and I can observe tabs changing and execute this methods. What do you think?